source/common/config/delta_subscription_state.cc

```diff
@@ -124,7 +124,9 @@ void DeltaSubscriptionState::handleGoodResponse(
   {
     const auto scoped_update = ttl_.scopedTtlUpdate();
     for (const auto& resource : message.resources()) {
-      addResourceState(resource);
+      if (wildcard_ || resource_state_.contains(resource.name())) {
```

Comment:
I think these resources should not be inserted into the `non_heartbeat_resources` (see above).

Reply:
we talked offline, the complication here is that we don't want to adds too much change in this CL(e.g. xdstp glob matching), we will create another issue to track the refactoring work instead, possibly align with the ongoing effort of unifying the SotW and Delta-xDS subscription work.